src/Storefront/Controller/EsiController.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Controller;
+
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Shopware\Storefront\Pagelet\Footer\FooterPageletLoader;
+use Shopware\Storefront\Pagelet\Header\HeaderPageletLoaderInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * @internal
+ */
+#[Route(defaults: ['_routeScope' => ['storefront']])]
+#[Package('storefront')]
+class EsiController extends StorefrontController
```

Comment:
I would suggest, to not introduce one global ESI controller. instead having a controller for each header and footer. and then additional controller for the ESI things to come in the future. 

or is there any advantage having all route in one (going to be big) controller class?

Reply:
I would suggest to not introduce special endpoints for esi at all. render_esi should just call the endpoint that calls the pageletLoader. If this is handled with esi or not is an configuration detail.

Reply:
But right now we have no controller endpoint which "just calls" the pagelet loader.

Reply:
Then implement one

Reply:
My point is, that there is no need to do a dedicated ESI controller if we can have the same (same number of lines and code) just with a name that don't force a special kind of layer. A HeaderController and FooterController with the same methods and not "esi" in its path does not add more code or complexity but more freedom of the exact kind of implementation.

Reply:
But keep in mind, the route has to be marked as `_esi`

```
#[Route(path: '/esi/header', name: 'frontend.esi.header', defaults: ['_httpCache' => true, '_esi' => true], methods: ['GET'])]
```

Which also means, this route should only be called via ESI because the sub request handling is another then a normal request

Reply:
But it could also called by something "like esi" for instance if we implement some kind of defered loading, or would there be a problem then?